### PR TITLE
tidb: update the default value of variable `tidb_enable_non_prepared_plan_cache`

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

TiDB PR: https://github.com/pingcap/tidb/pull/43904

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by chrysan — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -1623,7 +1623,7 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
 - 类型：布尔型
-- 默认值：`OFF`
+- 默认值：`ON`
```
Comment (qiancai) [inline system-variables.md:1626]: 这里改成了 ON，但是前面提到 “非 Prepare 语句执行计划缓存目前为实验特性，不建议在生产环境中使用。”，是希望用户手动改成 OFF 吗

Comment (qiancai) [reply system-variables.md:1626]: @songrijie 请确认下 `tidb_enable_non_prepared_plan_cache` 从 OFF 改成 ON，是打算 GA 吗？

Comment (songrijie) [reply system-variables.md:1626]: 这里并不打算GA，是想在DMR默认打开，收集反馈，争取在LTS前GA

**Review by songrijie — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]
/lgtm

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**